### PR TITLE
Replace NOT operator with explicit `false` check

### PR DESCRIPTION
**Pull request by pugnascotia**
We have an in-house rule to compare explicitly against `false` instead
of using the logical not operator (`!`). However, this hasn't
historically been enforced, meaning that there are many violations in
the source at present.

We now have a Checkstyle rule that can detect these cases, but before we
can turn it on, we need to fix the existing violations. This is being
done over a series of PRs, since there are a lot to fix.

### CODE REVIEW

File: libs/core/src/main/java/org/elasticsearch/common/Booleans.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public static boolean parseBooleanLenient(String value, boolean defaultValue) {
         if (value == null) {
             return defaultValue;
         }
-        return !(value.equals("false") || value.equals("0") || value.equals("off") || value.equals("no"));
+        return (value.equals("false") || value.equals("0") || value.equals("off") || value.equals("no")) == false;
```
Comment (pgomulka) [inline libs/core/src/main/java/org/elasticsearch/common/Booleans.java:131]: this probably requires a little bit head scratching. Would it be easier to read if it was series of && and each being == false? Also not sure this needs to be addressed here.

**Review by pgomulka — APPROVED** [review]
LGTM, some of the expressions are not super obvious, but they were not obvious before either (the once with double negations or alternatives)

File: libs/core/src/main/java/org/elasticsearch/common/Booleans.java (hunk 1 above)
Comment (pugnascotia) [reply libs/core/src/main/java/org/elasticsearch/common/Booleans.java]: I refactored it to a `switch`.

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by pugnascotia** [comment]
Backported to `7.x` in e8da7e33fd2

**Pull request merged**